core/modules/ccontrol/UserQuerySelect.cc

```diff
@@ -217,35 +217,27 @@ void UserQuerySelect::submit() {
         auto& chunkSpec = *i;
         auto cs = _qSession->buildChunkQuerySpec(queryTemplates, chunkSpec);
         auto endQSpecQSJ = std::chrono::system_clock::now(); // TEMPORARY-timing
-        chunks.push_back(cs.chunkId);
-        std::string chunkResultName = ttn.make(cs.chunkId);
+        chunks.push_back(cs->chunkId);
+        std::string chunkResultName = ttn.make(cs->chunkId);
         ++msgCount;
         auto endChunkPushQSJ = std::chrono::system_clock::now(); // TEMPORARY-timing
-        std::ostringstream ss;
-        taskMsgFactory.serializeMsg(cs, chunkResultName, _executive->getId(), sequence, ss);
-        std::string msg = ss.str();
 
-        proto::ProtoImporter<proto::TaskMsg> pi;
-        if (!pi.messageAcceptable(msg)) {
-            throw UserQueryBug(getQueryIdString() + " Error serializing TaskMsg.");
-        }
-        auto endChunkSerializeQSJ = std::chrono::system_clock::now(); // TEMPORARY-timing
-
-        std::shared_ptr<ChunkMsgReceiver> cmr = ChunkMsgReceiver::newInstance(cs.chunkId, _messageStore);
+        std::shared_ptr<ChunkMsgReceiver> cmr = ChunkMsgReceiver::newInstance(cs->chunkId, _messageStore);
         ResourceUnit ru;
-        ru.setAsDbChunk(cs.db, cs.chunkId);
+        ru.setAsDbChunk(cs->db, cs->chunkId);
         auto endChunkResourceQSJ = std::chrono::system_clock::now(); // TEMPORARY-timing
-        qdisp::JobDescription jobDesc(sequence, ru, ss.str(),
-                std::make_shared<MergingHandler>(cmr, _infileMerger, chunkResultName));
+        qdisp::JobDescription::Ptr jobDesc = qdisp::JobDescription::newJobDescription(
```

Comment:
I agree that create is a better name than newJobDescription. I think JobDescription is not quite long enough to merit a rename.